lib/auth/apiserver_test.go

```diff
@@ -142,7 +143,18 @@ func TestUpsertServer(t *testing.T) {
 
 func TestUpsertUser(t *testing.T) {
```

Comment:
This test was actually the only test that took advantage of using `auth.ClientI` to inject a mock to replace ServerWithRoles. This doesn't feel like a strong argument to keep it around - I've rewritten this test to just use a real implementation instead of a mock. It's shorter and actually tests the code better imo.

Reply:
+1 for testing with real objects.